plugins/woocommerce/src/Admin/Features/PaymentGatewaySuggestions/Init.php

```diff
@@ -35,21 +35,29 @@ public function __construct() {
 	 * @return array
 	 */
 	public static function get_suggestions( array $specs = null ) {
-		$suggestions = array();
 		if ( null === $specs ) {
 			$specs = self::get_specs();
 		}
 
+		$suggestions = array();
+		$has_error   = false;
 		foreach ( $specs as $spec ) {
 			try {
 				$suggestion    = EvaluateSuggestion::evaluate( $spec );
 				$suggestions[] = $suggestion;
 				// phpcs:ignore Generic.CodeAnalysis.EmptyStatement.DetectedCatch
 			} catch ( \Throwable $e ) {
-				// Ignore errors.
+				$has_error = true;
 			}
 		}
 
+		if ( $has_error ) {
```

Comment:
What do you think we move this code into the catch block? I don't think we need $has_error.

Reply:
I originally wanted to do that too, but because it's a loop, we might end up calling `set_specs_transient` many times. 🤔

Reply:
Oh my bad 🤦 I did not see the loop.